l10n_ch_lsv_dd/wizard/lsv_export_wizard.py

```diff
@@ -0,0 +1,537 @@
+        else:
+            raise exceptions.ValidationError(
+                _('Unable to determine clearing number for account %s') %
+                bank_account.acc_number)
+
+        return clearing
+
+    def _get_communications(self, line):
+        ''' This method can be overloaded to fit your communication style '''
+        return ''
+
+    def _get_ref(self, payment_line):
+        if self._is_bvr_ref(payment_line.move_line_id.transaction_ref):
+            return payment_line.move_line_id.transaction_ref.replace(
+                ' ', '').rjust(27, '0'), 'A'
+        return '', 'B'  # If anyone uses IPI reference, get it here
+
+    def _is_bvr_ref(self, ref):
+        if not ref:
+            return False  # Empty is not valid
+        clean_ref = ref.replace(' ', '')
+        if not clean_ref.isdigit() or len(clean_ref) > 27:
+            return False
+        clean_ref = clean_ref.rjust(27, '0')  # Add zeros to the left
+        if not clean_ref == mod10r(clean_ref[0:26]):
+            return False
+
+        return True
+
+    def _get_treatment_date(self, prefered_type, line_mat_date,
+                            order_sched_date, name):
+        ''' Returns appropriate date according to payment_order and
+            payment_order_line data.
+            Raises an error if treatment date is > today+30 or < today-10
+        '''
+        requested_date = date.today()
+        if prefered_type == 'due':
+            tmp_date = datetime.strptime(
+                line_mat_date, DEFAULT_SERVER_DATE_FORMAT
+            ).date()
+            requested_date = tmp_date if tmp_date else requested_date
+
+        elif prefered_type == 'fixed':
+            tmp_date = datetime.strptime(
+                order_sched_date, DEFAULT_SERVER_DATE_FORMAT
+            ).date()
+            requested_date = tmp_date if tmp_date else requested_date
+
+        if requested_date > date.today() + timedelta(days=30) \
+                or requested_date < date.today() - timedelta(days=10):
+            raise exceptions.ValidationError(
+                _('Incorrect treatment date: %s for line with ref %s')
+                % (requested_date, name))
+
+        return requested_date
+
+    def _is_ch_li_iban(self, iban):
+        ''' Check if given iban is valid ch or li iban '''
+        IBAN_CHAR_MAP = {
+            "A": "10",
+            "B": "11",
+            "C": "12",
+            "D": "13",
+            "E": "14",
+            "F": "15",
+            "G": "16",
+            "H": "17",
+            "I": "18",
+            "J": "19",
+            "K": "20",
+            "L": "21",
+            "M": "22",
+            "N": "23",
+            "O": "24",
+            "P": "25",
+            "Q": "26",
+            "R": "27",
+            "S": "28",
+            "T": "29",
+            "U": "30",
+            "V": "31",
+            "W": "32",
+            "X": "33",
+            "Y": "34",
+            "Z": "35"}

```

Comment:
An idea to replace this could be:

    dict((string.uppercase[i], 10 + i) for i in range(26))